arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -85,15 +85,16 @@ class EditAndSyncFeaturesViewController: UIViewController {
             if let error = error {
                 self.presentAlert(error: error)
             } else {
-                if let featureServiceInfo = self.geodatabaseSyncTask?.featureServiceInfo,
-                    let map = self.mapView.map {
-                    for index in featureServiceInfo.layerInfos.indices.reversed() {
-                        let layerInfo = featureServiceInfo.layerInfos[index]
-                        // For each layer in the serice, add a layer to the map.
-                        if let layerURL = self.featureServiceURL?.appendingPathComponent(String(index)) {
-                            let featureTable = AGSServiceFeatureTable(url: layerURL)
+                guard let featureServiceInfo = self.geodatabaseSyncTask?.featureServiceInfo, let map = self.mapView.map else { return }
+                for index in featureServiceInfo.layerInfos.indices {
+                    // For each layer in the serice, add a layer to the map.
+                    guard let layerURL = self.featureServiceURL?.appendingPathComponent(String(index)) else { return }
+                    let featureTable = AGSServiceFeatureTable(url: layerURL)
+                    featureTable.load { error in
```

Comment:
This closure is capturing self strongly. Also, a loadable holds on to the completion handler, so the handler should store an unowned reference to the loadable to avoid a retain cycle.

However, does the order of the layers matter? There's no guarantee that the tables will load in the order that `load()` is called. If order matters, use `AGSLoadObjects` to load all of the tables before creating the feature layers.

Reply:
Originally, the order of the layers did matter but regardless, only the point layer is added now. But it seems like I have to load all the layers in order to determine which is the point layer.